src/plots/declarative.py

```diff
@@ -27,21 +32,63 @@ def create_figure(self):
             raise ValueError(
                 'Number of plots does not match the number inputted rows'
                 'and columns.')
-        
-        # Create figure and axes
-        self.fig, self.axes = plt.subplots(self.nrows, self.ncols, figsize=self.figsize,
-                                 sharex=self.sharex, sharey=self.sharey)
-        # Plotting dictionary
+
         plot_dict = {
             'scatter': self._scatter,
             'histogram': self._histogram,
             'line_plot': self._lineplot,
             'vertical_line': self._verticalline,
             'horizontal_line': self._horizontalline,
             'bar_plot': self._barplot,
-            'horizontal_bar': self._hbar
+            'horizontal_bar': self._hbar,
+            'map_scatter': self._map_scatter,
+            'map_gridded': self._map_gridded,
+            'map_contour': self._map_contour
         }
-        # Feature dictionary
+
+        gs = gridspec.GridSpec(self.nrows, self.ncols)
+        self.fig = plt.figure(figsize=self.figsize)
+
+        for i in range(0, self.nrows*self.ncols):
```

Comment:
for i, plotobj in enumerate(self.plot_list): or whatever the syntax is?

Reply:
Great suggestion., thanks.